src/runtime/heap/objcache.c

```diff
@@ -227,6 +228,15 @@ static u64 objcache_allocate(heap h, bytes size)
     return obj;
 }
 
+static inline bytes objcache_size(void)
```

Comment:
I wrote it like this because that's how locking is implemented in the id heap code. @wjhun do you happen to remember why you put the spinlock just after the struct id_heap and not in the struct itself?

Reply:
I don't recall exactly; it might have been some half-baked idea about id_heap being a kind of invariant base type that could be extended without needing to add additional types. I would be fine with moving the lock inside the struct, since none of this was implemented.

Reply:
Now the struct spinlock is inside the objcache struct, under #ifdef KERNEL